### PR TITLE
Mh fi join shopping list

### LINKED ISSUE #5: 5. As a user, I want to join an existing shopping list so I can share a shopping list with another person.
Tokens can be shared with other users to allow them to co-manage a given list.

Joining an existing list consists of the following:

* User enters an existing token
* Save the token to `localStorage`
* Show the user the existing list

AC:

* When the user does not already have a token in `localStorage`, on the onboarding/home screen, a simple form is displayed that allows the user to enter a token
* Entering the token and hitting submit saves the token to `localStorage`, effectively giving them joint control of the list
* On submit, show an error if the token does not exist
* If they get an error message, allow them to try again or create a new list

Wireframe: https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/welcome-screen.png

### PR DESCRIPTION
**Pull request by fab33150**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description
  On the welcome page we added a form that allows the user to enter in their token to gain access(if the token is a match in the database) to an already established list.  If the token that the user enters into the form does not exist then the user will see an error message on the welcome page notifying them that their token is incorrect and that they can try again or create a brand new list. 

The useContext API was key to completing this story as well

## Related Issue

closes #5 

## Acceptance Criteria
When the user does not already have a token in localStorage, on the onboarding/home screen, a simple form is displayed that allows the user to enter a token
Entering the token and hitting submit saves the token to localStorage, effectively giving them joint control of the list
On submit, show an error if the token does not exist
If they get an error message, allow them to try again or create a new list

## Type of Changes

|     | Type                       |

|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

image

### After

image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -9,28 +9,33 @@ import {
...
-      <Switch>
-        <Route exact path="/" component={Welcome} />
-        <PrivateRoute path="/list">
-          <List />
-        </PrivateRoute>
-        <PrivateRoute path="/add">
-          <AddItem />
-        </PrivateRoute>
-      </Switch>
+      <TokenProvider>
```
[the remark this answers is not in the document]
Comment (skillitzimberg) [reply src/App.js]: @MabelM1, I would have structured the `TokenContext`/`TokenProvider` logic similar to the way that the `ItemsContext` is done - `ItemsContext.js` returns `ItemsProvider`.

So maybe `TokenProvider.js` could contain the setup done in `TokenContext.js` & `useTokenHook.js` and export  `TokenProvider` and `useContext(TokenContext);` as `useTokenHook`. Then these could be imported like `import { TokenProvider, useTokenHook } from './TokenProvider';` wherever they're needed.

Basically just fewer files. But as I said, that's totally personal preference.
